### DISCUSSION TITLE
Deleted object from schema still exists in the new realm

### DISCUSSION THREAD

**Discussion by vardansargsyan92** [realm/realm-dotnet#2540] [Q&A] [answered]
Hello

I do some preprocessing to prepare my realm to sync and, I am making some objects embedded. Before switching to realm-sync, I need to write migrations for supporting that changes (I have active users). So I have the following changes.

I have added partition keys and renamed primary keys (Id to _id) in the my realm object

**_before_**
```
[PrimaryKey]
public string Id { get; set; } = Guid.NewGuid().ToString();
```

**_after_**
```
[PrimaryKey]
[MapTo("_id")]
public string Id { get; set; } = Guid.NewGuid().ToString();

[Required]
[MapTo("_partitionKey")]
public string PartitionKey { get; set; }

```

I have added a migration function supporting partition keys.
I have removed one class (let's say class A) from the schema and make it embedded (Also renamed class A to class A' to avoid [this](https://github.com/realm/realm-dotnet/discussions/2393))
I have added a migration function for embedded objects.

When I restore the older databases,  the migration functions work fine except for one thing.

Class A that I have already deleted from the schema, still exists in the already migrated realm file. I guess that partition keys somehow interrupt the deletion of the objects.

So any ideas about what happens here?

**Comment by papafe** [comment, marked as answer]
Hi @vardansargsyan92, can I ask you how do you check for the existence of the class A objects?

**Reply by vardansargsyan92** [reply]
HI @papafe. Sure. I do it with MongoDb realm studio and I see the the redundant collections (objects that I already removed from the schema)

**Reply by papafe** [reply]
That's definitely interesting... How do you remove the class A from the schema? Do you completely remove from your project or are you specifying the realm classes in the configuration?

**Reply by vardansargsyan92** [reply]
@papafe  I remove it completely. Then create a A* (different name to avoid orphaned objects) embedded object and move whole A's functionality to A*. Then I increase a schema version and add necessary migration function to copy A's data to A*. I have an experience with removing objects and making them embedded that I have discussed [here](https://github.com/realm/realm-dotnet/discussions/2393). But now, something weird is happening when I add partition keys and add an attribute `[MapTo("_id")]`  to my primary keys of objects.

**Reply by papafe** [reply]
@vardansargsyan92 I have been investigating this and it seems it is working as it is supposed to be. The issue here is that the .NET api at the moment does not have a method to delete an entire class and his schema, but I have created a ticket about it just now, [here](https://github.com/realm/realm-dotnet/issues/2543). I am sorry for the inconvenience!

**Reply by vardansargsyan92** [reply]
HI @papafe . I am happy to see that you have detected the issue. Thanks a lot :-)
